packages/text-field/src/TextFieldPresenter.js

```diff
@@ -92,21 +96,26 @@ export default class TextFieldPresenter extends Component {
 
   static defaultProps = {
     id: generatedId(),
-    type: "text"
+    type: "text",
+    stayFocused: false
   };
 
   state = {
-    focused: false
+    focused: this.props.stayFocused
   };
 
   handleFocus = event => {
-    this.setState({ focused: true });
-    if (this.props.onFocus) this.props.onFocus(event);
+    if (!this.props.stayFocused) {
+      this.setState({ focused: true });
+      if (this.props.onFocus) this.props.onFocus(event);
+    }
   };
 
   handleBlur = event => {
-    this.setState({ focused: false });
-    if (this.props.onBlur) this.props.onBlur(event);
+    if (!this.props.stayFocused) {
```

Comment:
See comment above about the separation of presentational logic and business logic.